cmd/vpn_client/app/pathcontroller/ping.go

```diff
@@ -21,15 +23,25 @@ type icmpPinger struct {
 	retries int
 }
 
-const protocolICMP = 1
-
 func (p icmpPinger) Ping(client net.IP) error {
 	var err error
 	for i := 0; i < 1+p.retries; i++ {
 		err = p.ping(client)
 		if err == nil {
 			break
 		}
+		if i == 0 {
+			go func() {
+				// send neighbor solicitation to speed up discovery the link-layer address of a neighbor
+				p.log.Info("sending neighbor solicitation", "ip", client.String())
+				err := p.neighborSolicitation(client)
+				if err != nil {
+					p.log.Info("neighbor solicitation failed", "error", err.Error())
+				} else {
+					p.log.Info("received neighbor advertisement")
+				}
+			}()
+		}
```

Comment:
I can only report what I have seen in tcpdump. The ping ICMP messages don't go over the bond device. I cannot explain why. It can take up to 60s until neighbour solicitation packages are appearing. Therefore it is triggered manually.
The manual triggering is only performed if the first ping fails under the assumption that neighbour solicitation may not have be performed.
Triggering it on the second ping retry is useless. If it does not help after the first ping, the cause for the failing ping must be something else. Anyway the pings are performed every few seconds, so the next manual triggering is only seconds away.